src/InputOutput/readers.jl

```diff
@@ -513,11 +515,24 @@ function read_field(reader::HDF5Reader, name::AbstractString)
                 staggering = Grids.CellFace()
             end
             space = Spaces.space(grid, staggering)
-        else
+        elseif haskey(attrs(obj), "space")
             space = read_space(reader, attrs(obj)["space"])
+        else
+            # if the there is no grid, then the field is on a PointSpace
+            lg_obj = reader.file["local_geometry_data/$name"]
+            # TODO: Is this array type correct? should we assume SingletonCommsContext?
+            ArrayType = ClimaComms.array_type(ClimaComms.device(reader.context))
+            lg_data = ArrayType(read(lg_obj))
+            # because it is a point space, the data layout of local_geometry_data is always DataF
+            lg_type = eval(Meta.parse(attrs(lg_obj)["value_type"]))
```

Comment:
I think that this allows for arbitrary code execution just by adding the desired code into an attribute of the HDF5 file, making it a large security vulnerability. A malicious actor could modify a restart file in ClimaAtmos to change the `value_type` of one of its fields to silently execute arbitrary code when a user restarts the simulation with that file. While this is unlikely to be a concern right now, I think we should avoid using this pattern here and address the problem in a future PR.

When we parse something that could come from users, we should always sanitize it and check it against the allowed inputs.